src/uu/wc/src/count_fast.rs

```diff
@@ -87,7 +92,28 @@ pub(crate) fn count_bytes_fast<T: WordCountable>(handle: &mut T) -> (usize, Opti
             // example with /proc/version and /sys/kernel/profiling. So,
             // if it is 0 we don't report that and instead do a full read.
             if (stat.st_mode as libc::mode_t & S_IFREG) != 0 && stat.st_size > 0 {
-                return (stat.st_size as usize, None);
+                // We perform an lseek here because its possible that we are given a fd that starts
+                // somewhere further like so:
+                //
+                // ```bash
+                // ~$ stat --format='%s' /etc/group
+                // 1088
+                // ~$ (dd ibs=2 skip=1 count=0; wc -c) < /etc/group
+                // 1086
+                // ```
+                //
+                // Therefore, we perform a lseek with 0 offset to know where we actually starts.
+                // Note that GNU currently returns 0 if lseek fails but I am not sure how that can
+                // happen.
+                let offset = is_stdin
+                    .then(|| lseek(fd, 0, nix::unistd::Whence::SeekCur).ok())
+                    .flatten()
+                    .unwrap_or(0);
+
+                #[cfg(target_os = "android")]
+                let offset = offset as i64;
+
+                return (stat.st_size.sub(offset).max(0) as usize, None);
```

Comment:
It is necessary because `lseek` is allowed to return an offset that is beyond the end of file...for fun maybe!

```
lseek() allows the file offset to be set beyond the end of the
file (but this does not change the size of the file).  If data is
later written at this point, subsequent reads of the data in the
gap (a "hole") return null bytes ('\0') until data is actually
written into the gap.
```

Reply:
Hmm interesting. Could you add that info in a comment in the code?

Reply:
Yea I added here https://github.com/uutils/coreutils/pull/3987/commits/b7b4ed8e7f5f63ddc77076c51ec4fcc61bdfdec2
had to scour the manpages since I forgot why I wrote it myself haha